src/components/Stations/components/StationTableColumns.tsx

```diff
@@ -0,0 +1,165 @@
+import BaseWeatherIcon from "@/components/BaseComponents/BaseWeatherIcon";
+import CommonButton from "@/components/Common/CommonButton";
+import SvgInline from "@/components/Common/SvgInline";
+import { Measurements, WeatherConditions, WeatherDataResponse } from "@/types";
+import { HeartIcon } from "@heroicons/react/24/solid";
+import { createColumnHelper, SortingFn } from "@tanstack/react-table";
+import { fullDateWithTime } from "../../../utils/dateTimeUtils";
+
+// Sorting Fns
+const sortStatusFn: SortingFn<WeatherDataResponse> = (rowA, rowB) => {
+    return rowA.original.weather_station_id.name.localeCompare(
+        rowB.original.weather_station_id.name
+    );
+};
+const sortTemperatureFn: SortingFn<WeatherDataResponse> = (rowA, rowB) => {
+    return rowA.original.temperature - rowB.original.temperature;
+};
+
+// Columns initialisation
+const columnHelper = createColumnHelper<WeatherDataResponse>();
+export const getColumns = (
+    title: string,
+    handleFavouriteButton: (stationId: number) => void,
+    isStationFavourite: (stationId: number) => boolean
+) => {
+    return [
+        columnHelper.accessor((row) => row.weather_station_id.id, {
+            id: "stationFavourite",
+            cell: (info) => {
+                return (
+                    <div>
+                        <CommonButton
+                            handleClick={() => handleFavouriteButton(info.getValue())}
+                            color={isStationFavourite(info.getValue()) ? "primary" : "secondary"}
+                        >
+                            <HeartIcon className="size-8 p-1" />
+                        </CommonButton>
+                    </div>
+                );
+            },
+            header: () => <span>Favourite</span>,
+            sortingFn: sortStatusFn,
```

Comment:
I would disable sorting to be honest for this column.

Reply:
yep makes sense, I've updated it